src/app/qgsattributetabledialog.cpp

```diff
@@ -159,6 +159,7 @@ QgsAttributeTableDialog::QgsAttributeTableDialog( QgsVectorLayer *layer, QgsAttr
 
   QgsFeatureRequest r;
   bool needsGeom = false;
+  initialMode = settings.enumValue( QStringLiteral( "qgis/attributeTableBehavior" ),  QgsAttributeTableFilterModel::ShowAll );
```

Comment:
This should be getting passed to the constructor from the code in QgisApp - setting it here will break the ctrl+f6/shift+f6 shortcuts.

Can you check the QgisApp code to see if that's what's broken?

Reply:
Opening the attribute table was using the default from **qgisapp.h** given by:
``void attributeTable( QgsAttributeTableFilterModel::FilterMode filter = QgsAttributeTableFilterModel::ShowAll );``

Although the parameter should have been given by ``` connect( mActionOpenTable, &QAction::triggered, this, [ = ]
  {
    QgsSettings settings;
    QgsAttributeTableFilterModel::FilterMode initialMode = settings.enumValue( QStringLiteral( "qgis/attributeTableBehavior" ),  QgsAttributeTableFilterModel::ShowAll );
    attributeTable( initialMode );
  } );``` in does not get propagated to the ``attributeTable`` function (and the parameter was always 0 (default from **qgisapp.h**).

I don't understand why (maybe there's a more elegant solution) but IMHO always giving the initialMode parameter and not letting the default values should cover it.

It now functions as it should and it does not break the shortcuts.
@nyalldawson Thanks for reviewing, hope it's good. Any other insights are much appreciated.